### PR TITLE
cmd/buildah: add "manifest create --amend"

### PR DESCRIPTION
**Pull request by nalind**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

When `buildah manifest create` is given the `--amend` flag and a list with the name that was specified for the to-be-created list already exists, just reuse the list.

Make the `--insecure` flag, if we see it, conflict with `--tls-verify`, but have the reverse of the effect that `--tls-verify` does, and teach the `buildah manifest` `add`, `create`, and `push` subcommands about it.

#### How to verify it

Extended integration test!

#### Which issue(s) this PR fixes:

Related to https://github.com/containers/podman/issues/15346.

#### Does this PR introduce a user-facing change?

```release-note
`buildah manifest create` now accepts a `--amend` flag.
```

### CODE REVIEW

**Review by flouthoc — APPROVED** [review]
LGTM @containers/buildah-maintainers PTAL

File: cmd/buildah/manifest.go
Comment (TomSweeneyRedHat) [inline cmd/buildah/manifest.go:105]: soft suggestion for consideration
```suggestion
	flags.BoolVar(&manifestCreateOpts.insecure, "insecure", false, "neither require HTTPS nor verification of certificates when accessing the registry. TLS verification cannot be used when talking to an insecure registry.")
```

Comment (TomSweeneyRedHat) [reply cmd/buildah/manifest.go:105]: Then down below too if you take it.

**Comment by TomSweeneyRedHat** [comment]
one small doc tweak for consideration or to be ignored, your call.

LGTM otherwise.

File: cmd/buildah/manifest.go
Comment (rhatdan) [reply cmd/buildah/manifest.go:105]: Since it is hidden, it really does not matter, nobody will ever see it.

**Pull request merged**
